src/backend/lib/redis.js

```diff
@@ -1,39 +1,8 @@
-require('./config');
-const Redis = require('ioredis');
-const MockRedis = require('ioredis-mock');
-const { logger } = require('../utils/logger');
-const parseUrl = require('../utils/url-parser');
-
-// If you need to set the Redis URL, do it in REDIS_URL
-const redisUrl =
-  parseUrl(process.env.REDIS_URL, process.env.REDIS_PORT) || 'redis://127.0.0.1:6379';
-
-// Set MOCK_REDIS=1 to mock, MOCK_REDIS= to use real redis
-const useMockRedis = process.env.MOCK_REDIS;
-
-// RedisConstructor is one of Redis or MockRedis
-const RedisConstructor = useMockRedis ? MockRedis : Redis;
-
-function createRedisClient() {
-  try {
-    const { port, host } = new URL(redisUrl);
-    return new RedisConstructor(port, host, { password: process.env.REDIS_PASSWORD });
-  } catch (error) {
-    const message = `Unable to parse port and host from "${redisUrl}"`;
-    logger.error({ error }, message);
-    throw new Error(message);
-  }
-}
-
-// If using MockRedis, shim info() until https://github.com/stipsan/ioredis-mock/issues/841 ships
-if (useMockRedis && typeof MockRedis.prototype.info !== 'function') {
-  logger.debug('Shimming MockRedis info() method');
-  MockRedis.prototype.info = () => Promise.resolve('redis_version:999.999.999');
-}
+const { Redis } = require('@senecacdot/satellite');
 
 module.exports = {
```

Comment:
You should initialize redis like this
```js
const redis = Redis();
```

After this, you can export redis like this
```js
module.exports = redis;
```

We deal with the initialization in the method itself so no need to redeclare createRedisClient and redis. This also means you'll need to edit the backend files that currently use these 2 objects

Reply:
You can also combine that:

```js
module.exports = Redis();
```

Reply:
I've tried exporting only the instance `redis`. Luckily, there's only `queue.js` that uses `createRedisClient`. The containers start normally but when I start the backend `pnpm start`, it has an error. 
```
[ 2021-12-16 16:54:56.411 ] ERROR (7444 on DESKTOP-P67GFS3): Error queuing feeds
    err: {
      "type": "Error",
      "message": "Connection in subscriber mode, only subscriber commands may be used",
      "stack":
          Error: Connection in subscriber mode, only subscriber commands may be used
              at Redis.sendCommand (D:\Repos\telescope\node_modules\.pnpm\ioredis@4.28.2\node_modules\ioredis\built\redis\index.js:633:24)
              at Redis.hgetall (D:\Repos\telescope\node_modules\.pnpm\ioredis@4.28.2\node_modules\ioredis\built\commander.js:122:25)
              at getFeed (D:\Repos\telescope\src\backend\utils\storage.js:61:26)
              at byId (D:\Repos\telescope\src\backend\data\feed.js:157:24)
              at Array.map (<anonymous>)
              at Function.all (D:\Repos\telescope\src\backend\data\feed.js:182:28)
              at processTicksAndRejections (node:internal/process/task_queues:96:5)
              at async Promise.all (index 0)
              at async processAllFeeds (D:\Repos\telescope\src\backend\index.js:78:25)
    }
```

Reply:
In `queue.js`, it works normally if I assign to `client` and `subscriber` different redis intances
Instead of this
```js
const redis = require('./redis');
const client = redis;
const subscriber = redis;
```
This works 
```js 
const { Redis } = require('@senecacdot/satellite');
const client = Redis();
const subscriber = Redis();
```

Reply:
So I think, we should export both `const redis = Redis()` instance and the function `Redis`

Reply:
Makes sense for how you're doing it.